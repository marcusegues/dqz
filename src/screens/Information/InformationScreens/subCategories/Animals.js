// @flow
import React from 'react';
// $FlowFixMe
import { Text, TouchableOpacity } from 'react-native';
import type { ComponentType } from 'react';
import { translate } from 'react-i18next';
// $FlowFixMe
import type { Navigation, TFunction } from '../../../../types/generalTypes';
import { verticalScale } from '../../../../styles/Scaling';
import { AppInfoSubText } from '../../../AppInfo/subComponents/AppInfoSubText';
import { CardRowText } from '../../../../components/QuestionAnswer/Cards/subcomponents/CardRowText';
import { SubCategoryContainer } from '../../subComponents/SubCategoryContainer';

type AnimalsInnerProps = {
  t: TFunction,
  navigation: Navigation,
};

const AnimalsInner = ({ t, navigation }: AnimalsInnerProps) => (
  <SubCategoryContainer title={t('animalsMainText')}>
    <AppInfoSubText
      text={t('animalsText1')}
      style={{ marginTop: verticalScale(10) }}
    />
    <TouchableOpacity onPress={() => {}}>
      <Text>http://www.blv.admin.ch</Text>
    </TouchableOpacity>

    <CardRowText
      text={t('animalsText2')}
      style={{ marginTop: verticalScale(20) }}
    />
    <AppInfoSubText text={t('animalsText3')} />
    <TouchableOpacity
      onPress={() =>
        navigation.dispatch({
          type: 'NAVIGATE',
          screen: 'VatAllowance',
        })
      }
    >
      <AppInfoSubText text={t('animalsText4')} />
    </TouchableOpacity>
  </SubCategoryContainer>
);

export const Animals = (translate(['animals'])(AnimalsInner): ComponentType<
  AnimalsInnerProps
>);
