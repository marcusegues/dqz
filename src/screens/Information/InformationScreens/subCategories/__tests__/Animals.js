import React from 'react';
// $FlowFixMe
import { TouchableOpacity } from 'react-native';
import { shallow } from 'enzyme';
import { Animals } from '../Animals';

jest.mock('react-i18next', () => ({
  translate: () => Component => props => <Component t={() => ''} {...props} />,
}));

const clickFn = jest.fn();

const wrapper = shallow(
  <Animals navigation={{ dispatch: clickFn }} i18n={{ language: 'de' }} />
);

describe('Animals', () => {
  test('Animals renders as expected', () => {
    expect(wrapper.dive()).toMatchSnapshot();
  });
  test('you can click it', () => {
    const render = wrapper.dive();
    render.find(TouchableOpacity).simulate('press');
    expect(wrapper.dive()).toMatchSnapshot();
    expect(clickFn.mock.calls.length).toBe(1);
  });
});
