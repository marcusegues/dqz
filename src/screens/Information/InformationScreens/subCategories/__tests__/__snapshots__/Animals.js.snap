// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animals Animals renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AnimalsInner
    i18n={
      Object {
        "language": "de",
      }
    }
    navigation={
      Object {
        "dispatch": [Function],
      }
    }
    t={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <AppInfoSubText
          greyBox={false}
          style={
            Object {
              "marginTop": 19.61764705882353,
            }
          }
          text=""
        />,
        <AppInfoLink
          onPress={[Function]}
          style={Object {}}
          text="http://www.blv.admin.ch"
        />,
        <CardRowText
          greyBox={false}
          style={
            Object {
              "marginTop": 39.23529411764706,
            }
          }
          text=""
        />,
        <AppInfoSubText
          greyBox={false}
          style={Object {}}
          text=""
        />,
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <AppInfoSubText
            greyBox={false}
            style={
              Object {
                "textDecorationLine": "underline",
                "textDecorationStyle": "solid",
              }
            }
            text=""
          />
        </TouchableOpacity>,
      ],
      "title": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {
            "marginTop": 19.61764705882353,
          },
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onPress": [Function],
          "style": Object {},
          "text": "http://www.blv.admin.ch",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {
            "marginTop": 39.23529411764706,
          },
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {},
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <AppInfoSubText
            greyBox={false}
            style={
              Object {
                "textDecorationLine": "underline",
                "textDecorationStyle": "solid",
              }
            }
            text=""
          />,
          "onPress": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {
              "textDecorationLine": "underline",
              "textDecorationStyle": "solid",
            },
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <AppInfoSubText
            greyBox={false}
            style={
              Object {
                "marginTop": 19.61764705882353,
              }
            }
            text=""
          />,
          <AppInfoLink
            onPress={[Function]}
            style={Object {}}
            text="http://www.blv.admin.ch"
          />,
          <CardRowText
            greyBox={false}
            style={
              Object {
                "marginTop": 39.23529411764706,
              }
            }
            text=""
          />,
          <AppInfoSubText
            greyBox={false}
            style={Object {}}
            text=""
          />,
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
          >
            <AppInfoSubText
              greyBox={false}
              style={
                Object {
                  "textDecorationLine": "underline",
                  "textDecorationStyle": "solid",
                }
              }
              text=""
            />
          </TouchableOpacity>,
        ],
        "title": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {
              "marginTop": 19.61764705882353,
            },
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onPress": [Function],
            "style": Object {},
            "text": "http://www.blv.admin.ch",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {
              "marginTop": 39.23529411764706,
            },
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {},
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <AppInfoSubText
              greyBox={false}
              style={
                Object {
                  "textDecorationLine": "underline",
                  "textDecorationStyle": "solid",
                }
              }
              text=""
            />,
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "greyBox": false,
              "style": Object {
                "textDecorationLine": "underline",
                "textDecorationStyle": "solid",
              },
              "text": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Animals you can click it 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AnimalsInner
    i18n={
      Object {
        "language": "de",
      }
    }
    navigation={
      Object {
        "dispatch": [Function],
      }
    }
    t={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <AppInfoSubText
          greyBox={false}
          style={
            Object {
              "marginTop": 19.61764705882353,
            }
          }
          text=""
        />,
        <AppInfoLink
          onPress={[Function]}
          style={Object {}}
          text="http://www.blv.admin.ch"
        />,
        <CardRowText
          greyBox={false}
          style={
            Object {
              "marginTop": 39.23529411764706,
            }
          }
          text=""
        />,
        <AppInfoSubText
          greyBox={false}
          style={Object {}}
          text=""
        />,
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <AppInfoSubText
            greyBox={false}
            style={
              Object {
                "textDecorationLine": "underline",
                "textDecorationStyle": "solid",
              }
            }
            text=""
          />
        </TouchableOpacity>,
      ],
      "title": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {
            "marginTop": 19.61764705882353,
          },
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onPress": [Function],
          "style": Object {},
          "text": "http://www.blv.admin.ch",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {
            "marginTop": 39.23529411764706,
          },
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "greyBox": false,
          "style": Object {},
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <AppInfoSubText
            greyBox={false}
            style={
              Object {
                "textDecorationLine": "underline",
                "textDecorationStyle": "solid",
              }
            }
            text=""
          />,
          "onPress": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {
              "textDecorationLine": "underline",
              "textDecorationStyle": "solid",
            },
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <AppInfoSubText
            greyBox={false}
            style={
              Object {
                "marginTop": 19.61764705882353,
              }
            }
            text=""
          />,
          <AppInfoLink
            onPress={[Function]}
            style={Object {}}
            text="http://www.blv.admin.ch"
          />,
          <CardRowText
            greyBox={false}
            style={
              Object {
                "marginTop": 39.23529411764706,
              }
            }
            text=""
          />,
          <AppInfoSubText
            greyBox={false}
            style={Object {}}
            text=""
          />,
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
          >
            <AppInfoSubText
              greyBox={false}
              style={
                Object {
                  "textDecorationLine": "underline",
                  "textDecorationStyle": "solid",
                }
              }
              text=""
            />
          </TouchableOpacity>,
        ],
        "title": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {
              "marginTop": 19.61764705882353,
            },
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onPress": [Function],
            "style": Object {},
            "text": "http://www.blv.admin.ch",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {
              "marginTop": 39.23529411764706,
            },
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "greyBox": false,
            "style": Object {},
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <AppInfoSubText
              greyBox={false}
              style={
                Object {
                  "textDecorationLine": "underline",
                  "textDecorationStyle": "solid",
                }
              }
              text=""
            />,
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "greyBox": false,
              "style": Object {
                "textDecorationLine": "underline",
                "textDecorationStyle": "solid",
              },
              "text": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
